pkg/oc/cli/cmd/set/triggers.go

```diff
@@ -435,7 +429,25 @@ func (o *TriggersOptions) updateTriggers(triggers *TriggerDefinition) {
 		if o.FromBitbucket != nil && *o.FromBitbucket {
 			triggers.BitbucketWebHooks = nil
 		}
-		return
+	}
+
+	if len(o.FromImage) > 0 {
+		var updatedTriggers []ImageChangeTrigger
+		targetContainerNames := strings.Split(o.ContainerNames, ",")
+		for _, trigger := range triggers.ImageChange {
+			// If containers names specified matches the target container names
+			// then reset the trigger for this container
+			if sets.NewString(trigger.Names...).HasAny(targetContainerNames...) {
```

Comment:
This is wrong for cases where you have multiple containers for trigger specified in containerNames. You will overwrite the old image with the new one even for the rest of containers that it wasn't suppose to. I guess the right action would be to remove such container(s) from containersNames (if there are any remaining) and creating a new trigger.

Reply:
```
  imageChangeParams:
    containerNames: ["c1", "c2"]
    from:
      kind: ImageStreamTag
      name: "centos:latest"
```
```
oc set triggers dc/test-deployment --containers=c1 --from-image=ubuntu:latest
```
It should switch only c1's image to ubuntu and leave c2's image alone (to be still centos). `HasAny` looks like it will get overwritten for both. I don't see code splitting this trigger into 2 anywhere.